nilearn/plotting/img_plotting.py

```diff
@@ -60,42 +60,68 @@ def show():
 # Core, usage-agnostic functions
 
 
-def _get_colorbar_and_data_ranges(stat_map_data, vmax, symmetric_cbar, kwargs,
-                                  force_min_stat_map_value=None):
+def _get_colorbar_and_data_ranges(
+    stat_map_data, vmin=None, vmax=None, symmetric_cbar=True,
+    force_min_stat_map_value=None, symmetric_data_range=True,
+):
     """Set colormap and colorbar limits.
 
-    Used by for plot_stat_map and plot_glass_brain.
-
-    The limits for the colormap will always be set to range from -vmax to vmax.
-    The limits for the colorbar depend on the symmetric_cbar argument, please
-    refer to docstring of plot_stat_map.
+    Used by plot_stat_map, plot_glass_brain and plot_img_on_surf.
 
+    If symmetric_data_range is True, the limits for the colormap will
+    always be set to range from -vmax to vmax. The limits for the colorbar
+    depend on the symmetric_cbar argument, please refer to docstring of
+    plot_stat_map.
     """
-    if 'vmin' in kwargs:
+    if symmetric_data_range and (vmin is not None):
         raise ValueError('this function does not accept a "vmin" '
                          'argument, as it uses a symmetrical range '
                          'defined via the vmax argument. To threshold '
                          'the plotted map, use the "threshold" argument')
 
-    # make sure that the color range is symmetrical
-    if vmax is None or symmetric_cbar in ['auto', False]:
-        # Avoid dealing with masked_array:
-        if hasattr(stat_map_data, '_mask'):
-            stat_map_data = np.asarray(
-                stat_map_data[np.logical_not(stat_map_data._mask)])
-        stat_map_max = np.nanmax(stat_map_data)
-        if force_min_stat_map_value is None:
-            stat_map_min = np.nanmin(stat_map_data)
-        else:
-            stat_map_min = force_min_stat_map_value
+    # avoid dealing with masked_array:
+    if hasattr(stat_map_data, '_mask'):
+        stat_map_data = np.asarray(
+            stat_map_data[np.logical_not(stat_map_data._mask)])
+
+    if force_min_stat_map_value is None:
+        stat_map_min = np.nanmin(stat_map_data)
+    else:
+        stat_map_min = force_min_stat_map_value
+    stat_map_max = np.nanmax(stat_map_data)
+
+    # check compatibility between vmin, vmax and symmetric_cbar
+    if symmetric_cbar or symmetric_data_range:
+        if vmin is None and vmax is None:
+            vmax = max(-stat_map_min, stat_map_max)
+            vmin = -vmax
+        elif vmin is None:
+            vmin = -vmax
+        elif vmax is None:
+            vmax = -vmin
+        elif not np.isclose(vmin, -vmax):
+            raise ValueError(
+                "vmin must be equal to -vmax unless symmetric_cbar"
```

Comment:
I added it because I wasn't sure how to handle the case where both `vmin` and `vmax` are given (i.e. not `None`) but they are not the same and `symmetric_cbar` is `True`. The options I can think of are to:
1. force the colourbar to be symmetric by changing `vmin` or `vmax`, maybe with a warning that the user input is being overridden
2. ignore `symmetric_cbar` and just use the user-provided `vmin` and `vmax` (also with a warning)
3. not allow this scenario to happen (current behaviour)

What do you think?

To clarify, this is only when the user provides values for `vmin` and `vmax` and `symmetric_cbar=True`. If only `vmin` or `vmax` is provided then that's not a problem (the colourbar can still be symmetric). Maybe the error message isn't clear enough, I can change it if that's the case.

Reply:
> I'm wondering: do we require vmin=-vmax is symmetric_cbar is True ?

As I understand it we do, this is what is defined in the docstrings of the plotting functions that use this parameter (e.g. see [plot_surf_stat_map](https://nilearn.github.io/stable/modules/generated/nilearn.plotting.plot_surf_stat_map.html#nilearn-plotting-plot-surf-stat-map))

However, I think symmetric_data_range should not be in the error message as this is used internally and not exposed to the user (in fact in the current implementation the user can pass `symmetric_data_range=True` and `symmetric_cbar=False` with defined unequal vmin/vmax values and the error is not raised because the value of symmetric_data_range is overridden internally)

As I see it symmetric_data_range is only for controlling internal behavior with regards to other functions that do not yet allow vmin like `plot_glass_brain` and this parameter is meant to be temporary until the other functions are dealt with (in other PRs). @michellewang is this correct?

Last thing, the functions that use cbar_vmin/cbar_vmax only use them if symmetric_cbar is not True, otherwise they are set to None and eventually set to the normalized vmin/vmax and they are only used internally

Reply:
> However, I think symmetric_data_range should not be in the error message as this is used internally and not exposed to the user (in fact in the current implementation the user can pass symmetric_data_range=True and symmetric_cbar=False with defined unequal vmin/vmax values and the error is not raised because the value of symmetric_data_range is overridden internally)

That makes sense, I'll change the error message

> As I see it symmetric_data_range is only for controlling internal behavior with regards to other functions that do not yet allow vmin like plot_glass_brain and this parameter is meant to be temporary until the other functions are dealt with (in other PRs). @michellewang is this correct?

Yes exactly